crates/wasi/src/preview2/pipe.rs

```diff
@@ -273,28 +302,48 @@ impl AsyncWriteStream {
         }
     }
 
-    fn send(&mut self, bytes: Bytes) -> anyhow::Result<(usize, StreamState)> {
-        use tokio::sync::mpsc::error::TrySendError;
+    fn send(&mut self, msg: WriteMesage) -> anyhow::Result<bool> {
+        assert!(!self.has_pending_op());
```

Comment:
Nit: is this necessary if you have to check for the pending op anyway down below?